components/wmtk_components/marching/internal/Marching.cpp

```diff
@@ -0,0 +1,112 @@
+#include "Marching.hpp"
+
+#include <wmtk/SimplicialComplex.hpp>
+#include <wmtk/operations/tri_mesh/EdgeCollapseToMidpoint.hpp>
+#include <wmtk/operations/tri_mesh/EdgeSplitAtMidpoint.hpp>
+#include <wmtk/operations/tri_mesh/EdgeSplitWithTag.hpp>
+#include <wmtk/operations/tri_mesh/EdgeSwap.hpp>
+#include <wmtk/operations/tri_mesh/FaceSplitWithTag.hpp>
+#include <wmtk/operations/tri_mesh/VertexTangentialLaplacianSmooth.hpp>
+
+namespace wmtk::components::internal {
+
+Marching::Marching(
+    TriMesh& mesh,
+    MeshAttributeHandle<double>& position_handle,
+    MeshAttributeHandle<long>& vertex_tag,
+    MeshAttributeHandle<long>& edge_tag,
+    const long input_tag_value,
+    const long embedding_tag_value,
+    const long split_tag_value,
+    const int dimension,
+    const bool lock_boundary)
+    : m_mesh(mesh)
+    , m_position_handle(position_handle)
+    , m_vertex_tag(vertex_tag)
+    , m_edge_tag(edge_tag)
+    , m_input_tag_value(input_tag_value)
+    , m_embedding_tag_value(embedding_tag_value)
+    , m_split_tag_value(split_tag_value)
+    , m_dimension(dimension)
+    , m_scheduler(mesh)
+    , m_lock_boundary(lock_boundary)
+{}
```

Comment:
A `dimension = 0` never makes sense. Marching requires `dimension` to be at least 1 or greater. You can add an assert here to make sure of that already in the constructor.